File: AtCoder/ABC/ABC051-100/abc080/abc080c.py
n = int(input())
ans = float("-INF")
f = [[0 for i in range(10)] for i in range(n)]
p = [[0 for i in range(11)] for i in range(n)]
for i in range(n):
  ft = list(map(int, input().split()))
  for j in range(10):
    f[i][j] = ft[j]
for i in range(n):
  pt = list(map(int, input().split()))
  for j in range(11):
    p[i][j] = pt[j]

for i in range(1, 2**10):
  t = 0
  for j in range(n):
    c = 0
    for k in range(10):
      if i>>k&1 and f[j][k]:
        c += 1
    t += p[j][c]
  ans = max(ans, t)
print(ans)

File: AtCoder/ABC/ABC051-100/abc080/abc080c.cpp
#include <bits/stdc++.h>
using namespace std;
#define REP(i,b,e) for(int i=(b);i<(e);++i)
#define RREP(i,b,e) for(int i=(b)-1;i>=e;--i)
#define rep(i,e) for(int i=0;i<(e);++i)

constexpr int MOD = 1000000007;
constexpr int INF = 1LL << 30;
constexpr long long LLINF = 1LL << 62;

inline void print(void) { cout<<'\n'; }
template<class T> inline void print(const T &x) { cout<<x<<'\n'; }
template<class T, class... U> inline void print(const T &x, const U&... y) { cout<<x<<" "; print(y...); }

int f[110][10];
int p[110][11];

int main() {
  int n; cin>>n;
  rep(i, n) rep(j, 10) cin>>f[i][j];
  rep(i, n) rep(j, 11) cin>>p[i][j];

  int ans = -INF;
  REP(i, 1, 1<<10) {
    int ret = 0;
    rep(j, n) {
      int cnt = 0;
      rep(k, 10) if (i & (1<<k) and f[j][k]) ++cnt;
      ret += p[j][cnt];
    }
    ans = max(ans, ret);
  }
  print(ans);

  return 0;
}
